pkg/test/resourcefixture/contexts/register.go

```diff
@@ -95,41 +92,39 @@ func GetResourceContext(fixture resourcefixture.ResourceFixture, serviceMetadata
 		rc.ResourceGVK = fixture.GVK
 	}
 
-	// Determine the type of controller for the resource: TF, DCL, or Direct.
-	crd, err := crdloader.GetCRDForGVK(rc.ResourceGVK)
-	if err != nil {
-		return ResourceContext{}, err
-	}
-	if crd.GetLabels()[crdgeneration.TF2CRDLabel] == "true" {
-		rc.IsTFResource = true
-	} else if crd.GetLabels()[crdgeneration.Dcl2CRDLabel] == "true" {
-		rc.IsDCLResource = true
-	} else {
-		rc.IsDirectResource = true
-	}
-
 	if rc.RecreateDelay == 0 {
 		// By default, wait for 10 seconds before recreating resource as part of drift detection test.
 		// Some resources appear to need to wait longer than 10 seconds, so this value is customizable.
 		rc.RecreateDelay = time.Second * 10
 	}
 
-	if rc.IsDCLResource {
+	// If CRD has DCL controller label, fetch DCL schema.
+	crd, err := crdloader.GetCRDForGVK(rc.ResourceGVK)
+	if err != nil {
+		return ResourceContext{}, err
+	}
+	if crd.GetLabels()[crdgeneration.Dcl2CRDLabel] == "true" {
 		s, err := dclschemaloader.GetDCLSchemaForGVK(rc.ResourceGVK, serviceMetadataLoader, dclSchemaLoader)
 		if err != nil {
 			panic(fmt.Sprintf("error getting the DCL schema for GVK %v: %v", rc.ResourceGVK, err))
 		}
 		rc.DCLSchema = s
 	}
+
 	return rc, nil
 }
 
-func (rc ResourceContext) SupportsLabels(smLoader *servicemappingloader.ServiceMappingLoader) bool {
-	if rc.IsTFResource {
+func (rc ResourceContext) SupportsLabels(smLoader *servicemappingloader.ServiceMappingLoader, u *unstructured.Unstructured) bool {
+	rt, err := testreconciler.ReconcilerTypeForObject(u)
```

Comment:
I am not quite sure there is actually an API call caused by this? We are reading the CRDs from disk, so I felt it was ok to call repeatedly.

Reply:
You're right! Sorry I misread the function (there are two GetCRDForGVK() in crdloader.go).

Though I'm still not a fan loading all CRDs (more than 300 files) and unmarshaling each into a CRD object repetitively.

Reply:
Ok I have updated this change to use the in-memory GVK map to avoid repetitively loading the CRDs. Though, we still need to load the CRD each time we reconcile in the tests (for the schema updater), but this is the same as before. In the future, maybe we can explore some ways to improve (or remove?) the schema updater so that we would be able to avoid this.